sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -80,6 +80,20 @@ export class Sender {
     return sender.send(message);
   }
 
+  /**
+   * Opens the AMQP link to Azure Service Bus from the sender.
+   *
+   * It is not necessary to call this method in order to use the sender. It is
+   * recommended to call this before your first send() or sendBatch() call if you
+   * want to front load the work of setting up the AMQP link to the service.
+   */
+  open() {
+    // the underlying `MessageSender` initialization will both
+    // cache the sender in the context for the next send and
+    // ensure the link and connection are initialized.
+    return MessageSender.create(this._context).open();
```

Comment:
`MessageSender.create()` can throw an error, and `MessageSender.create().open()` can reject with an error. That creates a complication when it comes to handling errors from this method because unless you're using async/await you have to both wrap the method in a try/catch and attach a `catch` handler to the result.

Quick fix would be to make this method `async`, or do the try/catch around `MessageSender.create` on your own and wrap the error in a `Promise.reject`.

Reply:
Fixed - just made the method async but still returned the underlying promise from .open().

Reply:
What is making us say that `MessageSender.create()` can throw an error? Is it a general observation that methods can throw or is there something specific about this error?

All this does is instantiate the `MessageSender` class and update the cache in the client entity context. From what I know, nothing in this code path throws.

Reply:
The first line of [MessageSender.create](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/src/core/messageSender.ts#L824) does validation and can throw an error.

Reply:
lol, I have no idea how I missed that!